src/components/cw721-token/services/cw721-token.service.ts

```diff
@@ -83,61 +97,66 @@ export class Cw721TokenService {
-    let url: string = INDEXER_API.GET_NFTS_BY_OWNER;
-    const params = [
-      request.account_address,
-      this.indexerChainId,
-      CONTRACT_TYPE.CW721,
-      request.limit,
-    ];
+    const cw721Attributes = `id
+      token_id
+      owner
+      media_info
+      burned
+      cw721_contract {
+        name
+        smart_contract {
+          name
+          address
+        }
+      }`;
+
+    let whereClause: any = {
+      owner: { _eq: request?.account_address },
+      burned: { _eq: false },
+    };
+
     if (request?.keyword) {
-      url += '&%s=%s';
       if (
         request.keyword.startsWith(AURA_INFO.CONTRACT_ADDRESS) &&
         request.keyword.length === LENGTH.CONTRACT_ADDRESS
       ) {
-        params.push(SEARCH_KEYWORD.CONTRACT_ADDRESS);
+        whereClause = {
+          ...whereClause,
+          cw721_contract: {
+            smart_contract: { address: { _eq: request.keyword } },
+          },
+        };
       } else {
-        params.push(SEARCH_KEYWORD.TOKEN_ID);
+        whereClause = {
+          ...whereClause,
+          token_id: { _eq: request.keyword },
+        };
       }
-      params.push(encodeURIComponent(request.keyword));
     }
+
     if (request?.next_key) {
-      url += '&%s=%s';
-      params.push(SEARCH_KEYWORD.NEXT_KEY);
-      params.push(request.next_key);
-    }
-    const result = await this.serviceUtil.getDataAPI(
-      `${this.indexerUrl}${util.format(url, ...params)}`,
-      '',
-      ctx,
-    );
-    const tokens = result.data.assets.CW721.asset;
-    const count = result.data.assets.CW721.count;
-    const nextKey = result.data.nextKey;
-    if (tokens.length > 0) {
-      const listContractAddress = [
-        ...new Set(tokens.map((i) => i.contract_address)),
-      ];
-      const tokensInfo =
-        await this.smartContractRepository.getTokensByListContractAddress(
-          listContractAddress,
-        );
-      tokens.forEach((item) => {
-        item.token_name = '';
-        item.symbol = '';
-        item.decimals = 0;
-        const filter = tokensInfo.filter(
-          (f) => String(f.contract_address) === item.contract_address,
-        );
-        if (filter?.length > 0) {
-          item.token_name = filter[0].token_name;
-          item.symbol = filter[0].symbol;
-          item.decimals = filter[0].decimals;
-        }
-      });
+      whereClause = {
+        ...whereClause,
+        id: { _gt: request.next_key },
+      };
     }
 
-    return { tokens: tokens, count: count, next_key: nextKey };
+    const graphqlQuery = {
+      query: util.format(
+        INDEXER_API_V2.GRAPH_QL.CW721_OWNER,
+        this.chainDB,
+        cw721Attributes,
+      ),
+      variables: {
+        whereClause: whereClause,
+        limit: request?.limit,
+      },
+    };
+
+    const tokens = (await this.serviceUtil.fetchDataFromGraphQL(graphqlQuery))
+      .data[this.chainDB]['cw721_token'];
+    const count = tokens?.length;
```

Comment:
chỗ này bên kia limit thì count còn đúng k a

Reply:
count này trả về cho FE tự tính toán số lượng phần tử. Mặc định limit là 100 phân trang tới đâu cộng đến đấy.
![image](https://github.com/aura-nw/aura-explorer-api/assets/119287881/8cff1290-26d9-4122-9e87-31d4f96f0ea5)